src/components/utils/rqb-inputs/directory-item-selector.js

```diff
@@ -0,0 +1,324 @@
+                icon: getFileIcon(e.type, styles.icon),
+                children:
+                    e.type === elementType.DIRECTORY
+                        ? convertChildren(e.children)
+                        : undefined,
+                childrenCount:
+                    e.type === elementType.DIRECTORY
+                        ? e.subdirectoriesCount
+                        : undefined,
+            };
+        });
+    }, []);
+
+    const convertRoots = useCallback(
+        (newRoots) => {
+            return newRoots.map((e) => {
+                return {
+                    id: e.elementUuid,
+                    name: e.elementName,
+                    icon: getFileIcon(e.type, styles.icon),
+                    children:
+                        e.type === elementType.DIRECTORY
+                            ? convertChildren(
+                                  nodeMap.current[e.elementUuid].children
+                              )
+                            : undefined,
+                    childrenCount:
+                        e.type === elementType.DIRECTORY
+                            ? e.subdirectoriesCount
+                            : undefined,
+                };
+            });
+        },
+        [convertChildren]
+    );
+
+    const addToDirectory = useCallback(
+        (nodeId, content) => {
+            let [nrs, mdr] = updatedTree(
+                rootsRef.current,
+                nodeMap.current,
+                nodeId,
+                content
+            );
+            setRootDirectories(nrs);
+            nodeMap.current = mdr;
+            setData(convertRoots(nrs));
+        },
+        [convertRoots]
+    );
+
+    const updateRootDirectories = useCallback(() => {
+        fetchRootFolders(props?.types)
+            .then((data) => {
+                let [nrs, mdr] = updatedTree(
+                    rootsRef.current,
+                    nodeMap.current,
+                    null,
+                    data
+                );
+                setRootDirectories(nrs);
+                nodeMap.current = mdr;
+                setData(convertRoots(nrs));
+            })
+            .catch((error) => {
+                snackError({
+                    messageTxt: error.message,
+                    headerId: 'DirectoryItemSelector',
+                });
+            });
+    }, [convertRoots, props.types, snackError]);
+
+    useEffect(() => {
+        if (props.open) {
+            updateRootDirectories();
+        }
+    }, [props.open, updateRootDirectories]);
+
+    const fetchDirectory = useCallback(
+        (nodeId) => {
+            fetchDirectoryContent(nodeId, types)
+                .then((children) => {
+                    const childrenMatchedTypes = children.filter((item) =>
+                        contentFilter().has(item.type)
+                    );
+
+                    fetchElementsInfos(
+                        childrenMatchedTypes.map((e) => e.elementUuid)
```

Comment:
You should pass the FILTER type to the rest call, to have a better feedback when the user click on a folder: there will be an indication if a subfolder contains one or more elements of the given type (FILTER in your case). This is how its done in study-app.

Reply:
I also noticed this issue I think, we can't see that a directory contains a filter before clicking on the directory:
![pick_filter](https://github.com/gridsuite/gridexplore-app/assets/123802855/753a7256-f6cb-40b7-b54a-1239a1ebad29)